presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/HdfsFileIo.java

```diff
@@ -44,26 +41,24 @@ public HdfsFileIo(HdfsEnvironment environment, HdfsContext context)
     @Override
     public InputFile newInputFile(String path)
     {
-        Configuration configuration = environment.getConfiguration(context, new Path(path));
-        return HadoopInputFile.fromLocation(path, configuration);
+        return new PrestoHadoopInputFile(new Path(path), environment, context);
     }
 
     @Override
     public OutputFile newOutputFile(String path)
     {
-        Configuration configuration = environment.getConfiguration(context, new Path(path));
-        return HadoopOutputFile.fromPath(new Path(path), configuration);
+        return new PrestoHadoopOutputFile(new Path(path), environment, context);
     }
 
     @Override
     public void deleteFile(String pathString)
     {
         Path path = new Path(pathString);
         try {
-            environment.getFileSystem(context, path).delete(path, false);
+            environment.doAs(context.getIdentity().getUser(), () -> environment.getFileSystem(context, path).delete(path, false));
```

Comment:
Is this needed? We don't other delete calls.

Reply:
I think whether we should wrap a call is not super clear, and it'd be safer to wrap everything than not. We had a discussion in https://prestosql.slack.com/archives/CP1MUNEUX/p1593662755069900.

Reply:
I believe for delete operation it is not required, but +1 for be consistent for all operations.